### PR TITLE
feat: use @typescript-eslint for TypeScript rules

### PR DESCRIPTION
**Pull request by honzahovorka**
Since eslint/typescript-eslint-parser is deprecated, let's use @typescript-eslint/parser instead.

### CODE REVIEW

File: environments/typescript/recommended.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ module.exports = {
   },
 
   plugins: [
-    'typescript',
+    '@typescript-eslint',
```
Comment (robertrossmann) [inline environments/typescript/recommended.js:29]: Please add this as a dependency of this project. I think it is missing by accident; all the other plugins which are used ie. for React environment *are* bundled with this ruleset so TS plugin should also be bundled. 🙏 Thanks!

File: environments/typescript/recommended.js [hunk 2]
```diff
@@ -49,93 +49,100 @@ module.exports = {
...
+    '@typescript-eslint/interface-name-prefix': ['warn', 'always'],
 
     // Require a specific member delimiter style for interfaces and type literals
     // This rule aims to standardise the way interface and type literal members are delimited.
-    'typescript/member-delimiter-style': ['warn', {
-      delimiter: 'none',
+    '@typescript-eslint/member-delimiter-style': ['warn', {
+      multiline: {
+        delimiter: 'none',
+        requireLast: false
```
Comment (robertrossmann) [inline environments/typescript/recommended.js]: Is this change necessary in order to switch the packages or is this a rule change suggestion? 😇 If it is the latter I would prefer to have it in a separate PR so we can discuss the change with the whole team. 🙏

**Review by robertrossmann — CHANGES_REQUESTED** [review]
Let's add the TS plugin to the _package.json_ and remove rule behaviour modifications. 🙏 Otherwise, thanks for making the switch! 👍

File: environments/typescript/recommended.js (hunk 1 above)
Comment (honzahovorka) [reply environments/typescript/recommended.js:29]: Done 👍

File: environments/typescript/recommended.js (hunk 2 above)
Comment (honzahovorka) [reply environments/typescript/recommended.js]: It's a necessary change. It should behave in the same way as before 😇

**Review by robertrossmann — APPROVED** [review]
w00t! 🎉
